thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datalayer/bao/jdbc/TaskManagerImpl.java

```diff
@@ -103,6 +123,7 @@ public boolean updateStatusAndWorkerId(Long workerId, Long id, Set<TaskStatus> p
     if (permittedOldStatus.contains(task.getStatus())) {
       task.setStatus(newStatus);
       task.setWorkerId(workerId);
+      task.setStartTime(System.currentTimeMillis());
```

Comment:
Why setting startTime to current?

Reply:
Because this method is only used when workers pick up the task, we can update the start time within the same SQL statement. By doing so, we can ensure that all tasks picked up by workers have startTime and avoid additional DB call to update the start time.